popup.js

```diff
@@ -1,111 +1,157 @@
-//loading buttons
-var searchTextArea = document.getElementById('search');
+var searchTextArea = document.getElementById("search");
 var add = document.getElementById("add");
 var save = document.getElementById("download");
-var clear = document.getElementById('clear');
-var textarea = document.getElementById('textarea');
-var index = 0;
-var buttons = `<button class="tooltip tilebutton searchTile"><img src="https://img.icons8.com/material-outlined/15/000000/search--v2.png"/></a>
-<span class="tooltiptext">very soon!</span></button>
-<button class="tooltip tilebutton deleteTile"><img src="https://img.icons8.com/material-rounded/15/000000/delete.png"/></a>
-<span class="tooltiptext">soon!</span></button>`;
-
-
-//restoring localStorage
+var clear = document.getElementById("clear");
+var textarea = document.getElementById("textarea");
 var saved = localStorage.getItem('listItems');
-var note = localStorage.getItem('note');
+var list = document.getElementById('list');
+var index = 0;
 
 if (saved) {
 	list.innerHTML = saved;
+  assignButton();
 }
 
+// tells background.js that it opened
+chrome.runtime.sendMessage({text: "popup opened"});
 
 //add written note to list
-add.addEventListener('click', function(event) { 
-  addItemToList();
+add.addEventListener("click", function () {
+    addItemToList(document.getElementById('textarea').value);
```

Comment:
Do you need to query `document.getElementById('textarea')` again here? Can't you just reference `textarea` from above?